docs/docs/rad-lab-ui/ui_configure/ui-meta.md

```diff
@@ -61,10 +67,56 @@ variable "boot_disk_size_gb" {
 ```
 
 ## Updatesafe
+
 After a module has been deployed (the project created and resources created within it), some Terraform updates can be **destructive**. To warn the users that they may be performing a potentially destructive action, the `updatesafe` key word is added (or omitted) to the `UIMeta`.
 
 Any time a user makes an update to an existing deployment, the `updatesafe` values are checked. If **ANY** of the changed variables do **NOT** have the `updatesafe` keyword, the user will be warned that they likely will lose data by applying the update. Put another way, only if **ALL** the changed variables are `updatesafe` will the prompt be skipped.
 
+## Required
+
+Some of the variables are required to have value for deployment. If no `default` value works, ensure you set the `required` UIMeta keyword. Users will then be required to provide a value before deploying.
+
+**(Example)** `billing_account_id` will be shown as required for users and User must have to provide the input.
+
+```terraform
+variable "billing_account_id" {
+  description = "Billing Account associated to the GCP Resources. {{UIMeta group=0 order=3 required }}"
+  type        = string
+}
+```
+
+## Depends On
+
+Some of the variables are depends on other variables input, in order to make it dependent of the variables is determined by the `dependson` keyword. So based on user input and depends on condition satified to be shown the dependent variables.
+
+**(Example)** `container_image_repository` will be shown once `create_container_image` value will be true, similarly `ip_cidr_range` will be shown once `create_network` and `create_usermanaged_notebook` value will be true , similarly for other variables it will work based on `dependson` conditions with `||` and `&&` operands.
+
```

Comment:
```suggestion
**(Example)** `container_image_repository` will be shown once `create_container_image` value will be true, similarly `ip_cidr_range` will be shown once `create_network` and `create_usermanaged_notebook` value will be true , similarly for other variables it will work based on `dependson` conditions with `||` and `&&` operands.

It is recommended to put dependent variables immediately (or soon) after their parent variable by using the UIMeta `group` and `order` variables. This helps make a more logical, consistent flow for users.
```